lib/modules/versioning/maven/compare.spec.ts

```diff
@@ -276,11 +276,57 @@ describe('modules/versioning/maven/compare', () => {
         ],
       },
       {
-        input: 'alpha.beta.rc',
+        input: 'v1.2.3',
         expected: [
+          {
+            isTransition: false,
+            prefix: 'v',
+            type: 'TYPE_NUMBER',
+            val: 1,
+          },
+          {
+            isTransition: false,
+            prefix: 'PREFIX_DOT',
+            type: 'TYPE_NUMBER',
+            val: 2,
+          },
+          {
+            isTransition: false,
+            prefix: 'PREFIX_DOT',
+            type: 'TYPE_NUMBER',
+            val: 3,
+          },
+        ],
+      },
+      {
+        input: 'version2.3',
+        expected: [
+          {
+            isTransition: true,
+            prefix: '',
+            type: 'TYPE_QUALIFIER',
+            val: 'version',
+          },
           {
             isTransition: false,
             prefix: 'PREFIX_HYPHEN',
```

Comment:
won't it be no prefix, before 2?

Reply:
I didn't touch the logic for tokens beside the first token. Do it is hyphen prefix on 2 with isTransition true on the previous token. Probably because for the ordering `a1` and `a-1` are identical. Probably according to Maven specs the `v` should be treated as an own token, not a prefix. But the code just like before treats the leading v specially and assumes that both sides have it have not the v prefix. If you have one version with and one without v, the new code misbehaves the same as the old one. But treating it as token would need even deeper changes, as the major version would not any longer be the first token and so on. Thus I decided to just treat it as prefix like before just to not loose it on reconstruction.